### PR TITLE
Patch sarama_kafka rebalance fix

### PR DESCRIPTION
**Pull request by nbajaj90**
Issue: https://github.com/cloudevents/sdk-go/issues/817
Issue Explanation and Fix: https://github.com/Shopify/sarama/issues/2118
Fix reference: https://github.com/Shopify/sarama/blob/5e2c2ef0e429f895c86152189f625bfdad7d3452/examples/consumergroup/main.go#L177

### CODE REVIEW

**Comment by duglin** [comment]
LGTM
ping @lionelvillard for final review

File: protocol/kafka_sarama/v2/receiver.go
Comment (embano1) [inline protocol/kafka_sarama/v2/receiver.go:72]: nit: I generally recommend the context error here instead of surpressing it, e.g. `return session.Context().Err()`

File: protocol/kafka_sarama/v2/receiver.go
Comment (embano1) [inline protocol/kafka_sarama/v2/receiver.go:60]: question: since the behavior is now different from the previous implementation (the previous one basically never returned IIUC), should we add a comment here to describe when `ConsumeClaim` returns and what `error` would be (if any)?

File: protocol/kafka_sarama/v2/receiver.go
Comment (lionelvillard) [reply protocol/kafka_sarama/v2/receiver.go:72]: +1

**Comment by lionelvillard** [comment]
LGTM

@nbajaj90 can you look at @embano1 comments?

File: protocol/kafka_sarama/v2/receiver.go
Comment (nbajaj90) [reply protocol/kafka_sarama/v2/receiver.go:82]: Added a comment in code for this, please review.

File: protocol/kafka_sarama/v2/receiver.go
Comment (nbajaj90) [reply protocol/kafka_sarama/v2/receiver.go:60]: Added a comment in code for this, please review.

File: protocol/kafka_sarama/v2/receiver.go
Comment (embano1) [reply protocol/kafka_sarama/v2/receiver.go:82]: You're the expert on the implementation so I think it's ok. Probably better to put that comment in the Method comment section so it's exported and doc-ed with the method?

**Comment by lionelvillard** [comment]
LGTM

waiting for @embano1 for final review.

File: protocol/kafka_sarama/v2/receiver.go
Comment (nbajaj90) [reply protocol/kafka_sarama/v2/receiver.go:82]: No, I am not an expert, just wanted that fix patched here. Let me move the comment.

**Review by embano1 — APPROVED** [review]

**Comment by embano1** [comment]
Approved

**Comment by duglin** [comment]
LGTM
Merging

**Pull request merged**
